integration/v3_grpc_test.go

```diff
@@ -575,24 +582,61 @@ func TestV3WatchMultipleEventsFromCurrentRevision(t *testing.T) {
 		t.Fatalf("wAPI.Watch error: %v", wErr)
 	}
 
-	if err := wStream.Send(&pb.WatchRequest{CreateRequest: &pb.WatchCreateRequest{Prefix: []byte("foo")}}); err != nil {
+	if err := wStream.Send(&pb.WatchRequest{CreateRequest: &pb.WatchCreateRequest{Prefix: []byte("foo"), StartRevision: startRev}}); err != nil {
 		t.Fatalf("wStream.Send error: %v", err)
 	}
 
 	kvc := pb.NewKVClient(clus.RandConn())
-	txn := pb.TxnRequest{}
-	for i := 0; i < 3; i++ {
-		ru := &pb.RequestUnion{}
-		ru.RequestPut = &pb.PutRequest{Key: []byte(fmt.Sprintf("foo%d", i)), Value: []byte("bar")}
-		txn.Success = append(txn.Success, ru)
-	}
+	var wevents []*storagepb.Event
 
-	tresp, err := kvc.Txn(context.Background(), &txn)
-	if err != nil {
-		t.Fatalf("kvc.Txn error: %v", err)
-	}
-	if !tresp.Succeeded {
-		t.Fatalf("kvc.Txn failed: %+v", tresp)
+	if startRev == 0 { // test synced
+		for i := 0; i < 3; i++ {
+			if _, err := kvc.Put(context.TODO(), &pb.PutRequest{Key: []byte(fmt.Sprintf("foo%d", i)), Value: []byte("bar")}); err != nil {
```

Comment:
would this work just as well with only txns or puts but not both? if so it would simplify the logic a bit